### PR TITLE
Add Python3 support

### PR DESCRIPTION
**Pull request by jakobj**

### CODE REVIEW

File: h5py_wrapper/lib.py [hunk 1]
```diff
@@ -56,3 +57,11 @@ def convert_numpy_types_in_dict(d):
             d[key] = float(value)
         elif isinstance(value, (np.bool_)):
             d[key] = bool(value)
+
+
+def convert_iterable_to_numpy_array(it):
+    array = np.array(it)
```
Comment (mschmidt87) [inline h5py_wrapper/lib.py]: This function needs some explanation, such as:
"""
Converts an iterable to a numpy array. If the elements of the iterable are strings, they are converted to `np.string_`.
"""

File: .travis.yml [hunk 2]
```diff
@@ -2,6 +2,7 @@ sudo: false
 language: python
 python:
   - "2.7"
+  - "3.4"
```
Comment (mschmidt87) [inline .travis.yml]: Why not 3.5?

Comment (jakobj) [inline .travis.yml]: Python 3.5 lead to strange package version issues with conda, I hence opted for 3.4.

**Comment by jakobj** [comment]
Thanks for the comment, I addressed the issues you raised so far.

File: .travis.yml (hunk 2 above)
Comment (mschmidt87) [inline .travis.yml]: Okay.

File: h5py_wrapper/wrapper.py [hunk 3]
```diff
@@ -266,10 +258,10 @@ def _load_dataset(f, lazy=False):
         if value_type == 'NoneType':
             return None
         else:
-            if (len(f.attrs.keys()) > 0 and
-                    'custom_shape' in f.attrs.keys()):
+            if (len(list(f.attrs.keys())) > 0 and
```
Comment (mschmidt87) [inline h5py_wrapper/wrapper.py]: Actually, the list conversion is not needed here. 
We can just leave the code to be `len(f.attrs.keys())`.

**Comment by jakobj** [comment]
I've now added a Python3.4 badge and adapted `setup.py`. I've not added Python3.5 since we don't have that in our CI setup so things might break for 3.5 without travis noticing.

File: .travis.yml (hunk 2 above)
Comment (mschmidt87) [inline .travis.yml]: Can you describe the errors with python 3.5 and conda? Can we use  something alse than conda? I use Python 3.5 with the wrapper and it runs fine, so I think we should try to include in the CI, if possible in any way.

**Comment by jakobj** [comment]
Removed the list conversion. We already agreed on sticking with 3.4, right? Could you have another look?

**Comment by mschmidt87** [comment]
Okay, I just opened #63 to remind us to update to 3.5 once this issue has been solved on the conda level.

**Comment by jakobj** [comment]
I've now squashed the commits to produce just two: one adds py3 to the wrapper, one to travis.

File: .travis.yml (hunk 2 above)
Comment (mschmidt87) [inline .travis.yml]: Opened #63 to remind us of this issue.

**Comment by mschmidt87** [comment]
Excellent. :+1:

**Pull request merged**